src/openforms/registrations/contrib/objects_api/models.py

```diff
@@ -195,3 +196,38 @@ def apply_defaults_to(self, options):
             options["payment_status_update_json"] = self.payment_status_update_json
 
         options.setdefault("auteur", "Aanvrager")
+
+
+class ObjectsAPIRegistrationData(models.Model):
+    """Holds the temporary data available when registering a submission to the Objects API.
+
+    When starting the submission registration, this model is first populated. The Objects API
+    registration variables can then safely make use of this model to provide their data.
+    """
+
+    submission = models.OneToOneField(
+        "submissions.Submission",
+        on_delete=models.CASCADE,
+        verbose_name=_("submission"),
+        help_text=_("The submission linked to the registration data."),
+        related_name="objects_api_registration_data",
+    )
+    pdf_url = models.URLField(
+        _("pdf url"),
+        help_text=_("The PDF URL of the document on the Documents API."),
+        blank=True,
+    )
+    csv_url = models.URLField(
+        _("csv url"),
+        help_text=_("The CSV URL of the document on the Documents API."),
+        blank=True,
+    )
+    attachment_urls = ArrayField(
+        models.URLField(
+            _("attachment url"), help_text=_("The attachment URL on the Documents API.")
+        ),
+        verbose_name=_("attachment urls"),
+        help_text=_("The list of attachment URLs on the Documents API."),
+        blank=True,
+        default=list,
+    )
```

Comment:
I think this is too naive, you lose the information of which file/component the Document URL belongs too.

I'd still prefer to have the variable of a file upload explicitly mapped into the schema, e.g. something like:

```yaml
type: object
properties:
  proofOfEmployment:
    type: string
    format: uri
  bankStatements:
    type: array
    items:
        type: string
        format: uri
```
so that you can make the (single) file upload "proof of employment" to `/proofOfEmployment` and the multi-file upload "bank statements" to `/bankStatements`

Possibly we do this in a separate PR

Reply:
Considering this requires thinking and probably some work, it will probably be part of the next PR